argmin-py/src/solver.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018-2023 argmin developers
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+// TODO: docs
+
+use pyo3::prelude::*;
+
+use argmin::{core, solver};
+
+use crate::{
+    problem::Problem,
+    types::{IterState, Scalar},
+};
+
+#[pyclass]
+#[derive(Clone)]
+pub enum Solver {
+    Newton,
+}
+
+pub enum DynamicSolver {
+    // NOTE: I tried using a Box<dyn Solver<> here, but Solver is not object safe.
+    Newton(solver::newton::Newton<Scalar>),
+}
+
+impl From<Solver> for DynamicSolver {
+    fn from(solver: Solver) -> Self {
+        match solver {
+            Solver::Newton => Self::Newton(solver::newton::Newton::new()),
+        }
+    }
+}
+
+impl core::Solver<Problem, IterState> for DynamicSolver {
+    // TODO: make this a trait method so we can return a dynamic
```

Comment:
Sounds good to me! We could have both for backwards compatibility, right? The default impl of the `name` method would then just return `self.NAME`.

Reply:
It seems I was able to solve two problems at once: When I remove the associated constant, `Solver` becomes object-safe, so we can create trait objects for it. Let me know if you objects.

Reply:
Nice! I thought the generics would also be a problem for object safety, but it's great if this isn't the case. Sounds good to me!